src/index.ts

```diff
@@ -103,14 +103,10 @@ function useResizeObserver<T>(
         }
       }
     });
-  }, []);
+  });
```

Comment:
Removing the dependency array is necessary, in case `ref` is passed as null, then later passed as a valid value.